opacus/layers/dp_lstm.py

```diff
@@ -349,8 +463,29 @@ def forward(
 
         hs = torch.cat(hs, dim=0)  # [L * P, B, H]
         cs = torch.cat(cs, dim=0)  # [L * P, B, H]
-        out = self._rearrange_batch_dim(x)
-        return out, (hs, cs)
+
+        if batch_sizes is not None:
+            seq_lengths = compute_seq_lengths(batch_sizes)
+            packed_data = pack_padded_sequence(
+                pad_sequence(x, batch_first=False), seq_lengths, batch_first=True
+            )[0]
+            out = PackedSequence(
+                packed_data, batch_sizes, sorted_indices, unsorted_indices
+            )
+        else:
+            out = self._rearrange_batch_dim(x)
+
+        return out, (
+            self._permute_hidden(hs, unsorted_indices),
+            self._permute_hidden(cs, unsorted_indices),
+        )
+
+    def _permute_hidden(
+        self, x: torch.Tensor, permutation: Optional[torch.Tensor] = None, dim: int = 1
+    ):
+        if permutation is None:
+            return x
+        return x.index_select(dim, permutation)
```

Comment:
Using this function is not a good idea. It makes a copy of the tensor! https://pytorch.org/docs/stable/generated/torch.index_select.html

Reply:
Good catch, I will fix that.